storage/src/index/mod.rs

```diff
@@ -260,79 +251,89 @@ mod tests {
         index.insert(b"key", 4);
 
         assert_eq!(
-            index.get_iter(b"key").copied().collect::<Vec<_>>(),
+            index.iter(b"key").copied().collect::<Vec<_>>(),
             vec![4, 3, 2, 1]
         );
         assert!(context.encode().contains("pruned_total 0"));
 
         // Test removing first value from the list.
-        let mut iter = index.remove_iter(b"key");
-        iter.remove(); // should be a no-op
-        assert_eq!(*iter.next().unwrap(), 4);
-        iter.remove();
-        iter.remove(); // should be a no-op
-        assert!(context.encode().contains("pruned_total 1"));
+        {
+            let mut iter = index.mut_iter(b"key");
+            iter.remove(); // should be a no-op
+            assert_eq!(*iter.next().unwrap(), 4);
+            iter.remove();
+            iter.remove(); // should be a no-op
+            assert!(context.encode().contains("pruned_total 1"));
+        }
 
         assert_eq!(
-            index.get_iter(b"key").copied().collect::<Vec<_>>(),
+            index.iter(b"key").copied().collect::<Vec<_>>(),
             vec![3, 2, 1]
         );
 
         index.insert(b"key", 4);
         assert_eq!(
-            index.get_iter(b"key").copied().collect::<Vec<_>>(),
+            index.iter(b"key").copied().collect::<Vec<_>>(),
             vec![4, 3, 2, 1]
         );
 
         // Test removing from the middle.
-        let mut iter = index.remove_iter(b"key");
-        assert_eq!(*iter.next().unwrap(), 4);
-        assert_eq!(*iter.next().unwrap(), 3);
-        assert_eq!(*iter.next().unwrap(), 2);
-        iter.remove();
-        iter.remove(); // should be a no-op
-        assert!(context.encode().contains("pruned_total 2"));
+        {
+            let mut iter = index.mut_iter(b"key");
+            assert_eq!(*iter.next().unwrap(), 4);
+            assert_eq!(*iter.next().unwrap(), 3);
+            assert_eq!(*iter.next().unwrap(), 2);
+            iter.remove();
+            iter.remove(); // should be a no-op
+            assert!(context.encode().contains("pruned_total 2"));
+        }
 
         assert_eq!(
-            index.get_iter(b"key").copied().collect::<Vec<_>>(),
-            vec![4, 3, 1]
+            index.iter(b"key").copied().collect::<Vec<_>>(),
+            vec![3, 4, 1]
```

Comment:
We now swap the end (other option would be a shift but that seems much more expensive to maintain `LRU`-like behavior).